javaslang/src/main/java/javaslang/Lazy.java

```diff
@@ -133,12 +135,13 @@ private Lazy(Supplier<? extends T> supplier) {
      */
     @Override
     public T get() {
+        // using a local var speeds up the double-check idiom by 25%, see Effective Java, Item 71
```

Comment:
I agree that we should remove the speed-up comment.

However, it is hard to recognize design patterns/idioms when looking at code. Code is just a textual projection of abstract ideas (very much like 3D to 2D/Screen projection).

I want a comment for developers why we added a local var instead of using the well-known classical double-check idiom.